src/nvim/api/vim.c

```diff
@@ -695,6 +695,56 @@ Dictionary nvim_get_color_map(void)
   return colors;
 }
 
+/// Get the mappings for the corresponding scope and mode
+///
+/// @param mode
+/// @param scope
+Array nvim_get_keymap(String mode, String scope)
+  FUNC_API_SINCE(2)
+{
+  Array mappings = ARRAY_DICT_INIT;
+  mapblock_T *current_maphash;
+
+  // Convert the string mode to the integer mode
+  // that is stored within each mapblock
+  int int_mode = get_map_mode((char_u **)(&(mode.data)), 0);
+  int buffer_local = 0;
```

Comment:
This is boolean: `const bool buffer_local = strcmp(scope.data, "buffer") == 0;`.